specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json

```diff
@@ -0,0 +1,2134 @@
+          "description": "The client entity that created the JIT request."
+        },
+        "UpdatedBy": {
+          "$ref": "#/definitions/ApplicationClientDetails",
+          "description": "The client entity that last updated the JIT request."
+        }
+      },
+      "required":  [
+        "ApplicationResourceId",
+        "JitAuthorizationPolicies",
+        "JitSchedulingPolicy"
+      ],
+      "description": "Information about JIT request properties"
+    },
+    "JitAuthorizationPolicies": {
+      "properties": {
+        "PrincipalId": {
+          "type": "string",
+          "description": "The the principal id that will be granted JIT access."
+        },
+        "RoleDefinitionId": {
+          "type": "string",
+          "description": "The role definition id that will be granted to the Principal."
+        }
+      },
+      "required": [
+        "PrincipalId",
+        "RoleDefinitionId"
+      ],
+      "description": "The JIT authorization policies."
+    },
+    "JitSchedulingPolicy": {
+      "properties": {
+        "Type": {
+          "$ref": "#/definitions/JitSchedulingType",
+          "description": "The type of JIT schedule."
+        },
+        "Duration": {
+          "type": "string",
+          "format": "duration",
+          "title": "The required duration of the JIT request."
+        },
+        "StartTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The start time of the request."
+        }
+      },
+      "required": [
+        "Type",
+        "Duration",
+        "StartTime"
+      ],
+      "description": "The JIT scheduling policies."
+    },
+    "ApplicationClientDetails": {
+      "properties": {
+        "Oid": {
+          "type": "string",
+          "description": "The client Oid."
+        },
+        "Puid": {
+          "type": "string",
+          "description": "The client Puid"
+        },
+        "ApplicationId": {
+          "type": "string",
+          "description": "The client application Id."
+        }
+      },
+      "description": "The application client details to track the entity creating/updating the managed app resource."
+    },
+    "JitRequestState": {
+      "description": "The JIT request state.",
+      "readOnly": true,
+      "enum": [
+        "NotSpecified",
+        "Pending",
+        "Approved",
+        "Denied",
+        "Failed",
+        "Canceled",
+        "Expired",
+        "Timeout"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "JitRequestState",
+        "modelAsString": true
+      }
+    },
+    "JitSchedulingType": {
+      "description": "The JIT request scheduling type.",
+      "readOnly": true,
+      "enum": [
+        "NotSpecified",
+        "Once",
+        "Recurring"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "JitSchedulingType",
+        "modelAsString": true
+      
```

Comment:
which of these properties do the user provide and which can be marked as readonly?